otter/convergence/composition.py

```diff
@@ -66,17 +68,23 @@ def json_to_LBConfigs(lbs_json):
     return lbd
 
 
-def get_desired_group_state(launch_config, desired):
+def get_desired_group_state(group_id, launch_config, desired,
+                            _prepare_lc=prepare_launch_config):
     """
     Create a :obj:`DesiredGroupState` from a launch config and desired
     number of servers.
 
+    :param str group_id: Scaling group ID
     :param dict launch_config: Group's launch config as per
         :obj:`otter.json_schema.group_schemas.launch_config`
     :param int desired: Group's desired capacity
+    :param callable _prepare_lc: Optional implementation of
+        `prepare_launch_config` useful for testing
     """
     lbs = json_to_LBConfigs(launch_config['args']['loadBalancers'])
+    server_lc = {'server': launch_config['args']['server']}
     desired_state = DesiredGroupState(
-        launch_config={'server': launch_config['args']['server']},
+        launch_configs=(_prepare_lc(group_id, server_lc)
+                        for _ in itertools.count()),
```

Comment:
My thoughts exactly (see PR comment), but let's have the discussion here by the code :)